src/Tribe/RSVP.php

```diff
@@ -927,10 +927,39 @@ public function save_ticket( $event_id, $ticket, $raw_data = array() ) {
 
 		update_post_meta( $ticket->ID, '_price', $ticket->price );
 
-		if ( isset( $raw_data['ticket_stock'] ) && trim( $raw_data['ticket_stock'] ) !== '' ) {
-			$stock = (int) $raw_data['ticket_stock'];
+		$data = Tribe__Utils__Array::get( $raw_data, 'tribe-ticket', array() );
+
+		$default_capacity = -1;
+
+		// Fetch capacity field, if we don't have it use default (defined above)
+		$data['capacity'] = trim( Tribe__Utils__Array::get( $data, 'capacity', $default_capacity ) );
+
+		// If empty we need to modify to -1
+		if ( '' === $data['capacity'] ) {
+			$data['capacity'] = -1;
+		}
+
+		// The only available value lower than zero is -1 which is unlimited
+		if ( 0 > $data['capacity'] ) {
+			$data['capacity'] = -1;
+		}
+
+		// Fetch the stock if defined, otherwise use Capacity field
+		$data['stock'] = Tribe__Utils__Array::get( $data, 'stock', $data['capacity'] );
```

Comment:
Not sure it's needed, but is there value in either performing a `trim()` (like we do in similar code above) or else softening the next condition to test using `empty()` vs strictly testing for an empty string?